job-dsls/jobs/replace_shared_libraries.groovy

```diff
@@ -0,0 +1,97 @@
+def replaceShared = '''
+@Library('jenkins-pipeline-shared-libraries')_
+
+pipeline {
+    agent {
+        label 'kie-rhel7 && !master'
+    }
+    options {
+        buildDiscarder logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '', numToKeepStr: '10')
+        timeout(time: 20, unit: 'MINUTES')
+    }
+    stages {
+        stage('Initialize') {
+            steps {
+                sh 'printenv'
+
+            }
+        }
+        stage('Replace projects') {
+            steps {
+                script {
+                    def sharedLibrariesProjects = ['kiegroup/droolsjbpm-build-bootstrap': ['Jenkinsfile', 'Jenkinsfile.nightly'], 'jboss-integration/rhba' : ['Jenkinsfile']]
+                    sharedLibrariesProjects.each { key, val ->
+                        println "[INFO] Treating project ${key}"
+                        def projectGroupName = util.getProjectGroupName(key) 
+                        dir("${projectGroupName[0]}_${projectGroupName[1]}") {
+                            withCredentials([usernamePassword(credentialsId: 'kie-ci', usernameVariable: 'USER', passwordVariable: 'PASS')]) {
+                                def encodedPass = URLEncoder.encode(PASS, 'UTF-8')
+                                sh "git clone -b ${BRANCH_NAME} https://${USER}:${encodedPass}@github.com/${key} ."
+                                
+                                val.each { file -> 
+                                    println "[INFO] Treating file ${file}"
+                                    sh """
+                                    sed -i "/^@Library('jenkins-pipeline-shared-libraries')_/c\\\\@Library('jenkins-pipeline-shared-libraries@${BRANCH_NAME}')_" ${file}
+                                    """
+                                    sh "git add ${file}"
+                                }
+                                
+                                def changes = sh(returnStdout: true, script: 'git status -s').split()
+                                println "changes ${changes}"
+                                if(changes.size() > 0) {
+                                    println 'There are changes'
+                                    sh 'git config user.email "emingora@redhat.com"'
+                                    sh 'git config user.name "Enrique Mingorance Cano"'
```

Comment:
Do we have a service account that we could use with these? Unless there's a human intervention we shouldn't use a person account, it could be misleading.